outputs/jira_test.go

```diff
@@ -0,0 +1,1074 @@
+		},
+		{
+			name:      "sad path (Failed to create meta project)",
+			jiraApi:   &JiraAPI{},
+			wantError: "Failed to create meta project",
+		},
+		{
+			name:           "sad path (Failed to create issuetype)",
+			jiraApi:        &JiraAPI{Issuetype: "Bug", ProjectKey: "project"},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			wantError:      "Failed to create issuetype",
+		},
+		{
+			name:           "sad path (Failed to create fields config)",
+			jiraApi:        &JiraAPI{ProjectKey: "project"},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			priorityList:   &[]jira.Priority{{Name: "High"}},
+			wantError:      "Failed to create fields config",
+		},
+		{
+			name:           "sad path (Failed to init issue)",
+			jiraApi:        &JiraAPI{ProjectKey: "project", Unknowns: map[string]string{"bad field": "bad field"}},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			priorityList:   &[]jira.Priority{{Name: "High"}},
+			wantError:      "key bad field is not found in the list of fields",
+		},
+		{
+			name:           "sad path (Failed to open issue)",
+			jiraApi:        &JiraAPI{ProjectKey: "project"},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			priorityList:   &[]jira.Priority{{Name: "High"}},
+			wantError:      "Failed to open issue",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mux := http.NewServeMux()
+			mux.HandleFunc("/rest/api/2/issue/createmeta", func(w http.ResponseWriter, r *http.Request) {
+				if test.wantError == "Failed to create meta project" {
+					w.WriteHeader(http.StatusNotFound)
+				} else {
+					fieldListJson, _ := json.Marshal(test.createMetaInfo)
+					_, _ = w.Write(fieldListJson)
+				}
+			})
+			mux.HandleFunc("/rest/api/2/field", func(w http.ResponseWriter, r *http.Request) {
+				fieldListJson, _ := json.Marshal(test.fieldList)
+				_, _ = w.Write(fieldListJson)
+			})
+			mux.HandleFunc("/rest/api/2/priority", func(w http.ResponseWriter, r *http.Request) {
+				if test.wantError == "Failed to create fields config" {
+
+				} else {
+					priorityListJson, _ := json.Marshal(test.priorityList)
+					_, _ = w.Write(priorityListJson)
+				}
+			})
+			mux.HandleFunc("/rest/api/2/issue", func(w http.ResponseWriter, r *http.Request) {
+				if test.wantError == "Failed to open issue" {
+					w.WriteHeader(http.StatusNotFound)
+					_, _ = w.Write([]byte("Failed to open issue"))
+				} else {
+					issueJson, _ := json.Marshal(test.issue)
+					_, _ = w.Write(issueJson)
+				}
+			})
+			ts := httptest.NewServer(mux)
+
+			savedCreateClient := createClient
```

Comment:
same note as before - assign/restore it once

Reply:
Done